Google.PowerShell.IntegrationTests/Storage/Storage.GcsObject.Tests.ps1

```diff
@@ -207,6 +223,132 @@ Describe "New-GcsObject" {
         $obj.Metadata["Content-Type"] | Should Be "image/jpeg"
         Remove-GcsObject $obj
     }
+
+    It "should upload an empty folder" {
+        $folderName = [System.IO.Path]::GetRandomFileName()
+        $testFolder = "$env:TEMP/$folderName"
+
+        try {
+            New-Item -ItemType Directory -Path $testFolder | Out-Null
+
+            $folder = New-GcsObject -Bucket $bucket -Folder $testFolder -Force
+
+            $folder.Name | Should Be "$folderName/"
+            $folder.Size | Should Be 0
+
+            $folderOnline = Get-GcsObject -Bucket $bucket -ObjectName "$folderName/"
+            $folderOnline.Name | Should Be "$folderName/"
+            $folderOnline.Size | Should Be 0
+        }
+        finally
+        {
+            Remove-Item $testFolder -Recurse -Force -ErrorAction Ignore
+        }
+    }
+
+    It "should upload a folder with slashes at the end of the path" {
```

Comment:
To be clearer, perhaps "should work for both forward and backslashes in the path"?